Library/Homebrew/download_strategy.rb

```diff
@@ -532,6 +532,82 @@ def _fetch
   end
 end
 
+# GitHubReleaseDownloadStrategy downloads tarballs from GitHub Release assets.
+# To use it, add ":using => GitHubReleaseDownloadStrategy" to the URL section
+# of your formula. This download strategy uses GitHub access tokens (in the
+# environment variables GITHUB_TOKEN) to sign the request.
+# This strategy is suitable for corporate use just like S3DownloadStrategy,
+# because it lets you use a private GttHub repository for internal distribution.
+# It works with public one, but in that case simply use CurlDownloadStrategy.
+class GitHubReleaseDownloadStrategy < CurlDownloadStrategy
+  require 'open-uri'
```

Comment:
Oh, I did not know such a usuful lib. I changed to use `utils/github`.
I also changed the environment varaible used to this same `HOMEBREW_GITHUB_API_TOKEN` .